New negative history tests

thejohnjansen opened the pull request:
Add tests for attempts to navigate to a specific URL - negative tests

thejohnjansen committed:
New negative history tests

Add tests for attempts to navigate to a specific URL - negative tests

domenic commented on html/browsers/history/the-history-interface/history_go_to_uri-1.html:
```diff
@@ -0,0 +1,18 @@
+      opener.pages.push(id);
+      opener.start_test_wait();
+      if (!opener.gone) {
+          history.go("history_entry.html");
```
The go() method only accepts numbers, and will attempt to convert this into a number and throw an error when it does.

thejohnjansen replied:
yeah, that's what the test is testing: that .go() will fail in this case. IE11 supports it, but Edge does not (nor does Chrome, FF, etc), so we wanted to write a test to make sure we don't regress this interoperability.

domenic replied:
Hmm I think I see. Some comments or assertion messages might help :)

thejohnjansen replied:
Fixed. I've added a comment.

thejohnjansen replied:
I fixed the build issue. You cool with merging?

thejohnjansen replied:
Not sure who can approve this PR, but is there anything else blocking the merge?

dwsjax replied:
yes

On Jul 29, 2016 3:03 PM, "John Jansen" <notifications@github.com> wrote:

> In html/browsers/history/the-history-interface/history_go_to_uri-1.html
> <https://github.com/w3c/web-platform-tests/pull/3306#discussion_r72844681>
> :
>
> > @@ -0,0 +1,18 @@
> > +<!doctype html>
> > +<script src="history.js"></script>
> > +<script>
> > +  onunload = function() {}
> > +
> > +  onload = function() {
> > +    if (!opener.started) {
> > +      queue_next();
> > +    } else {
> > +      opener.pages.push(id);
> > +      opener.start_test_wait();
> > +      if (!opener.gone) {
> > +          history.go("history_entry.html");
>
> I fixed the build issue. You cool with merging?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/w3c/web-platform-tests/pull/3306/files/5c17662599d62fe174c3984841c70fadbbfe4fa4#r72844681>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AH8lSTsCDpgb11H2w-ba9dKB30K0sn8uks5qak5ygaJpZM4JO7lz>
> .
>

thejohnjansen committed:
Add a comment because the test is confusing

We are conducting a negative test here, which is confusing. Adding a
comment to clarify.

thejohnjansen committed:
lint error with trailing whitespace

weird issue with lint, but OK.

domenic commented:
So I am not usually the reviewer for this category, but since I stuck my nose in to comment, I feel responsible. So let's get this merged.

I've confirmed this test passes in Chrome and Firefox, so it's not broken. Let me leave some comments on how to make the test output and metadata a bit better. After those are addressed we can merge.

domenic commented on html/browsers/history/the-history-interface/history_go_to_uri.html:
```diff
@@ -0,0 +1,29 @@
+<!doctype html>
+<title>history.go()</title>
```
Title needs to be a bit more descriptive, e.g. "history.go() negative tests".

domenic commented on html/browsers/history/the-history-interface/history_go_to_uri.html:
```diff
@@ -0,0 +1,29 @@
+<!doctype html>
+<title>history.go()</title>
+<script src="/resources/testharness.js"></script>
```
Optionally add `<link rel="author" title="Domenic Denicola" href="mailto:d@domenic.me">` (but with your info instead). Preferably add `<link rel="help" href="https://html.spec.whatwg.org/multipage/browsers.html#dom-history-go">` and `<meta charset="utf-8">`.

domenic commented on html/browsers/history/the-history-interface/history_go_to_uri.html:
```diff
@@ -0,0 +1,29 @@
+      ), 500);
+    }
+  );
+  t.step(function() {win = window.open("history_entry.html?urls=history_go_to_uri-1.html,history_forward-2.html");
```
Nit: Indentation and formatting seems broken on this line and the next

thejohnjansen replied:
I'm not sure which line you mean here, so I guessed. :-) I'm learning this formatting, it's different from what I'm used to.

domenic commented on html/browsers/history/the-history-interface/history_go_to_uri.html:
```diff
@@ -0,0 +1,29 @@
+<script src="/resources/testharnessreport.js"></script>
+<div id="log"></div>
+<script>
+  var t = async_test(undefined, {timeout:5000});
```
This test needs a name. Maybe "history.go() must not navigate to URLs given as its argument; instead it must treat them as 0"

zcorpan replied:
The name is taken from document.title if omitted, which is fine when it's just one test in a file.

domenic commented on html/browsers/history/the-history-interface/history_go_to_uri-1.html:
```diff
@@ -0,0 +1,23 @@
+        // This is meant to test that passing a string is not supported.
+        // According to the spec, the value passed to 'go' must be an int.
+        // Internet Explorer supports passing a string and will navigate
+        // to that Uri. This test will protect against regressing in
```
Nit: s/Uri/URL

thejohnjansen committed:
Code review feedback

I somehow missed this PR was ongoing - i thought it had merged for some
reason. But I came across it today and see I had some changes to make.
Apologies for letting this sit for so long.
